test/unit/test_estimator.py

```diff
@@ -82,3 +83,20 @@ def test_estimator_circuit_caching(self):
                 inputs = kwargs["inputs"]
                 self.assertDictEqual(inputs["circuits"], {psi4_id: psi4})
                 self.assertEqual(inputs["circuit_ids"], [psi4_id, psi1_id])
+
+    def test_unsupported_values_for_estimator_options(self):
+        """Test exception when options levels are not supported."""
+        options_bad = [
+            {"optimization_level": 3, "resilience_level": 4},
+            {"optimization_level": 4, "resilience_level": 3},
+        ]
+        with Session(
+            service=FakeRuntimeService(channel="ibm_quantum", token="abc"),
+            backend="common_backend",
+        ) as session:
+            circuit = QuantumCircuit(1, 1)
+            obs = SparsePauliOp.from_list([("I", 1)])
+            for bad_opt in options_bad:
+                inst = Estimator(session=session)
+                with self.assertRaises(ValueError):
+                    _ = inst.run(circuit, observables=obs, **bad_opt)
```

Comment:
@merav-aharoni Can you also add this message verification to the other tests that have the same `self.assertRaises`?

Reply:
I added this to all the new tests with the exception of `test_resilience_options` in `test_ibm_primitives.py`, because that has very diverse error messages. I will try to find a way to do that there as well. Was there any other place you think I missed?